### PR TITLE
#164047063 Users should be able to create their profiles

### PR DESCRIPTION
**Pull request by JumaKahiga**
#### Description
___
During registration, a user providers authentication details `username and email`; however, for Authors Haven's usability and user interaction, it needs to have user profiles. Users can view each other's profiles, but a user can only edit his/her own. In addition, a user may deactivate his/her profile, which would make it not visible to other users.

#### Type of Change
___
[x] New Feature

#### How has this been tested
___
[x] Unit tests

#### Checklist
___
[x] I have added tests that prove my feature is working

#### Pivotal Tracker
___
[164047063](https://www.pivotaltracker.com/story/show/164047063)

### CODE REVIEW

**Comment by JoanNgatia** [comment]
@JumaKahiga Please make this PR point to the `develop` branch and not the `master` branch

**Review by Kibetchirchir — COMMENTED** [review]
Good work 👍 . But check on tests

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -134,6 +135,7 @@ class UserSerializer(serializers.ModelSerializer):
         min_length=8,
         write_only=True
     )
+    #profile = ProfileSerializer()
```
Comment (JoanNgatia) [inline authors/apps/authentication/serializers.py]: Remove this commented out line

File: authors/apps/authentication/tests/test_api.py [hunk 2]
```diff
@@ -30,47 +30,6 @@ def test_register_user(self):
         self.assertEqual(result["user"]["email"], "caro@yahoo.com")
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
-    def test_register_user_existing_email(self):
```
Comment (JoanNgatia) [inline authors/apps/authentication/tests/test_api.py:33]: Do we need to remove these tests?

File: authors/apps/profiles/custom_validations.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+	ok_formats = [
+					".png",
+					".jpg",
+					".jpeg"
+				]
+				
+	avatar_name =str(avatar_name)
+
+	if not avatar_name.lower().endswith((*ok_formats,)):
```
Comment (JoanNgatia) [inline authors/apps/profiles/custom_validations.py:18]: This implementation is really good.

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/tests/test_api.py (hunk 2 above)
Comment (JumaKahiga) [reply authors/apps/authentication/tests/test_api.py:33]: These tests were duplicated in test_registration_error_messages.py

File: authors/apps/profiles/custom_validations.py (hunk 3 above)
Comment (JumaKahiga) [reply authors/apps/profiles/custom_validations.py:18]: Thank you.

File: authors/apps/authentication/serializers.py (hunk 1 above)
Comment (JumaKahiga) [reply authors/apps/authentication/serializers.py]: Thanks. Changed

**Review by JoanNgatia — APPROVED** [review]

**Pull request merged**